src/main/java/racingcar/model/CarsPriorityBlockingQueue.java

```diff
@@ -0,0 +1,67 @@
+package racingcar.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.PriorityBlockingQueue;
+import java.util.stream.Collectors;
+
+/**
+ * @author : jiffyin7@gmail.com
+ * @since : 24. 10. 25.
+ */
+public class CarsPriorityBlockingQueue implements Cars{
+  private final PriorityBlockingQueue<Car> cars;
+
+  private CarsPriorityBlockingQueue(List<Car> cars){
+    this.cars = new PriorityBlockingQueue<>(cars);
+  }
+
+  public static CarsPriorityBlockingQueue from (List<Car> cars) {
+    return new CarsPriorityBlockingQueue(cars);
+  }
+
+  @Override
+  public List<Car> findCarsByPosition(int position) {
+    List<Car> foundCars = new ArrayList<>();
+    List<Car> temporary = new ArrayList<>();
+    while (!cars.isEmpty()){
+      Car car = cars.poll();
+      temporary.add(car);
+      if (car.getPosition() < position){
+        break;
+      }
+      if (car.getPosition() == position){
+        foundCars.add(car);
+      }
+    }
+    cars.addAll(temporary);
+    return foundCars;
+  }
```

Comment:
```suggestion
@Override
public List<Car> findCarsByPosition(int position) {
    return cars.stream()
               .filter(car -> car.getPosition() == position)
               .collect(Collectors.toList());
}
```
while 문과 조건문이 여러 중첩된 로직은 지양해야합니다
스트림을  이용하여 간결하게 변경 가능합니다

Reply:
오 단순히 3줄 만으로 저 긴 코드줄이 압축되네요👍